src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -86,67 +104,658 @@ static DamageTable D_80BAA468 = {
+    func_80BA8DF4(this, globalCtx);
+}
+
+s32 EnBaba_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                            Actor* thisx) {
+    EnBaba* this = THIS;
+
+    if (limbIndex == 6) {
+        Matrix_InsertTranslation(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->unk_2E4.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->unk_2E4.x, MTXMODE_APPLY);
+        Matrix_InsertTranslation(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if (limbIndex == 5) {
+        Matrix_InsertXRotation_s(-this->unk_2EA.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->unk_2EA.x, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == 6) && (this->unk_1E2 != 0) && ((globalCtx->state.frames % 2) == 0)) {
+        Matrix_InsertTranslation(40.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == 5) || (limbIndex == 10) || (limbIndex == 14)) {
+        rot->y += (s16)(Math_SinS(this->unk_302[limbIndex]) * 200.0f);
+        rot->z += (s16)(Math_CosS(this->unk_326[limbIndex]) * 200.0f);
+    }
+
+    if (((this->unk_40C == 1) || (this->unk_40C == 3) || (this->unk_40C == 4) ||
+         (ENBABA_GET_C000(&this->actor) == ENBABA_C000_2)) &&
+        (limbIndex == 8)) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+void EnBaba_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnBaba* this = THIS;
+    Vec3f sp18 = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == 7) {
+        this->actor.focus.pos.x = this->actor.world.pos.x;
+        this->actor.focus.pos.y = this->actor.world.pos.y;
+        this->actor.focus.pos.z = this->actor.world.pos.z;
+        Matrix_MultiplyVector3fByState(&sp18, &this->actor.focus.pos);
+    }
+}
+
+void EnBaba_UnkDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
+}
+
+void EnBaba_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnBaba* this = THIS;
+    Vec3f sp4C;
+    Vec3f sp40;
+
+    if (this->unk_40A & 2) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+
+        func_8012C5B0(globalCtx->state.gfxCtx);
+
+        gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(&object_bba_Tex_0092A0));
+
+        func_801343C0(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                      EnBaba_OverrideLimbDraw, EnBaba_PostLimbDraw, EnBaba_UnkDraw, &this->actor);
+
+        if (this->unk_40A & 0x80) {
+            if ((this->unk_40C == 3) || (this->unk_40C == 4)) {
+                func_8012C2DC(globalCtx->state.gfxCtx);
+                sp4C.x = this->actor.world.pos.x + 20.0f;
+                sp4C.y = this->actor.world.pos.y;
+                sp4C.z = this->actor.world.pos.z + 20.0f;
+            } else {
+                sp4C = this->actor.world.pos;
+            }
+
+            sp40.x = sp40.y = sp40.z = 0.3f;
+            func_800BC620(&sp4C, &sp40, 0xFF, globalCtx);
```

Comment:
```suggestion
            func_800BC620(&sp4C, &sp40, 255, globalCtx);
```
assuming the include/functions.txt is right and that is alpha